### ISSUE TITLE
Suggestion: Rename RWops types and functions to make more sense

### ISSUE THREAD

**Issue by furious-programming** [libsdl-org/SDL#7455]
From the beginning, I don't like the name `RWops`, because it's a bunch of letters, which at first glance you don't know what they mean. Also, the formatting of these names does not follow the naming convention used in the rest of the library — `RwOps` would be more appropriate.

Since read/write operations are about reading and writing data to any stream, why not just call it `SDL_Stream`? If you do this, the clarity will be high, the purpose of the functions will be clearly visible. The functions would rename to:

```c
SDL_AllocRW        → SDL_AllocStream
SDL_FreeRW         → SDL_FreeStream
SDL_RWclose        → SDL_StreamClose
SDL_RWFromConstMem → SDL_StreamFromConstMem
SDL_RWFromFile     → SDL_StreamFromFile
SDL_RWFromFP       → SDL_StreamFromFP
SDL_RWFromMem      → SDL_StreamFromMem
SDL_RWread         → SDL_StreamRead
SDL_RWseek         → SDL_StreamSeek
SDL_RWtell         → SDL_StreamTell
SDL_RWwrite        → SDL_StreamWrite
```
What do you think about it? This issue is related to the #6569.

**Comment by slouken** [comment]
There are lots of kinds of streams, so I'd prefer something like IOStream, but @libsdl-org/a-team thoughts?

**Comment by furious-programming** [comment]
There is a base class in the Object Pascal standard library, which is a similar abstraction to `RWops`. Any other streams inherit from it, including streams for files, arbitrary memory buffers, strings etc.. Its name is just `TStream` ([Free Pascal](https://www.freepascal.org/docs-html/rtl/classes/tstream.html), [Delphi](https://docwiki.embarcadero.com/Libraries/Sydney/en/System.Classes.TStream)) — hence my suggestion to replace `RWops` simply with `Stream` since it also applies to arbitrary streams and data buffers.

I like your proposal — `IOStream` is clear and unambiguous.

**Comment by 1bsyl** [comment]
"RWops" also has the meaning of "this can be overloaded", e.g. calling SDL_CreateRW like https://github.com/libsdl-org/SDL/blob/main/docs/README-migration.md#sdl_rwopsh to re-implement SDL_RWFromFP()

(There is also SDL_AudioStream which is some king of IO. I thought it would be elegant to create AudioStream on top of RWops.
but OTHO, as soon as you'll try to grep/find to a stream function, you'll end up with much more occurrence. so let's not change that)

In short, I'm fine with the current name / don't mind much if that changes.

**Comment by slouken** [comment]
Let's go ahead and leave it as-is for now. @libsdl-org/a-team, feel free to weigh in if you like a different naming for these functions.
